tests/transforms/test_get_unitary_matrix.py

```diff
@@ -0,0 +1,215 @@
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import reduce
+import pytest
+import numpy as np
+from gate_data import I, X, H, S, CNOT
+import pennylane as qml
+
+from pennylane.transforms.get_unitary_matrix import get_unitary_matrix
+
+# test non-parametric single qubit gates
+nonparam_1qubit_op_list = [qml.PauliX, qml.PauliY, qml.PauliZ, qml.Hadamard, qml.S, qml.T, qml.SX]
+
+
+@pytest.mark.parametrize("op", nonparam_1qubit_op_list)
+@pytest.mark.parametrize("wire", [0, 1, 2])
+def test_get_unitary_matrix_nonparam_1qubit_ops(op, wire):
+
+    wires = [0, 1, 2]
+
+    def testcircuit(wire):
+        op(wires=wire)
+
+    get_matrix = get_unitary_matrix(testcircuit, wires)
+    matrix = get_matrix(wire)
+
+    if wire == 0:
+        expected_matrix = np.kron(op(wires=wire).matrix, np.eye(4))
+    if wire == 1:
+        expected_matrix = np.kron(np.eye(2), np.kron(op(wires=wire).matrix, np.eye(2)))
+    if wire == 2:
+        expected_matrix = np.kron(np.eye(4), op(wires=wire).matrix)
+
+    assert np.allclose(matrix, expected_matrix)
+
+
+# Test a circuit containing multiple gates
+def test_get_unitary_matrix_multiple_ops():
+
+    wires = ["a", "b", "c"]
+
+    def testcircuit():
+        qml.PauliX(wires="a")
+        qml.S(wires="b")
+        qml.Hadamard(wires="c")
+        qml.CNOT(wires=["b", "c"])
+
+    get_matrix = get_unitary_matrix(testcircuit, wires)
+    matrix = get_matrix()
+
+    expected_matrix = np.kron(I, CNOT) @ np.kron(X, np.kron(S, H))
+
+    assert np.allclose(matrix, expected_matrix)
+
+
+def test_get_unitary_matrix_more_ops():
+    # Same circuit as test_single_qubit_fusion_multiple_qubits() in test_single_qubit_fusion.py
+    wires = ["a", "b"]
+
+    def testcircuit():
+        qml.RZ(0.3, wires="a")
+        qml.RY(0.5, wires="a")
+        qml.Rot(0.1, 0.2, 0.3, wires="b")
+        qml.RX(0.1, wires="a")
+        qml.CNOT(wires=["b", "a"])
+        qml.SX(wires="b")
+        qml.S(wires="b")
+        qml.PhaseShift(0.3, wires="b")
+
+    op1 = np.kron(qml.RZ(0.3, wires="a").matrix, I)
+    op2 = np.kron(qml.RY(0.5, wires="a").matrix, I)
+    op3 = np.kron(I, qml.Rot(0.1, 0.2, 0.3, wires="b").matrix)
+    op4 = np.kron(qml.RX(0.1, wires="a").matrix, I)
+    op5 = np.array([[1, 0, 0, 0], [0, 0, 0, 1], [0, 0, 1, 0], [0, 1, 0, 0]])
+    op6 = np.kron(I, qml.SX(wires="b").matrix)
+    op7 = np.kron(I, S)
+    op8 = np.kron(I, qml.PhaseShift(0.3, wires="b").matrix)
```

Comment:
I have written so many blocks of code like this lately for one of my other PRs, very very glad to have this new transform :joy: